lib/chef/api_client.rb

```diff
@@ -124,14 +124,21 @@ def to_json(*a)
       Chef::JSONCompat.to_json(to_hash, *a)
     end
 
-    def self.json_create(o)
-      client = Chef::ApiClient.new
-      client.name(o["name"] || o["clientname"])
-      client.private_key(o["private_key"]) if o.key?("private_key")
-      client.public_key(o["public_key"])
-      client.admin(o["admin"])
-      client.validator(o["validator"])
-      client
+    class << self
```

Comment:
I don't love it either but it seemed to be the canonical way to use alias on class methods.  I supposed I could just do:

```
def self.json_create(o)
   Chef::APiClient.from_hash(o)
end
```